src/precice/impl/WatchPoint.cpp

```diff
@@ -131,12 +97,12 @@ void WatchPoint::exportPointData(
     return;
   }
 
-  PRECICE_ASSERT(_vertices.size() == _weights.size());
+  //PRECICE_ASSERT(_vertices.size() == _weights.size());
   _txtWriter.writeData("Time", time);
   // Export watch point coordinates
   Eigen::VectorXd coords = Eigen::VectorXd::Constant(_mesh->getDimensions(), 0.0);
-  for (size_t i = 0; i < _vertices.size(); i++) {
-    coords += _weights[i].weight * _vertices[i]->getCoords();
+  for (const auto elem : _interpolationElements) {
```

Comment:
```suggestion
  for (const auto& elem : _interpolationElements) {
```
No need to copy every element. Same below.